intake_geopandas/geopandas.py

```diff
@@ -70,28 +83,80 @@ def __init__(self, urlpath, bbox=None,
         self._bbox = bbox
         self._geopandas_kwargs = geopandas_kwargs or {}
         self._dataframe = None
+        self.storage_options = storage_options or {}
+
+        # warn if using fsspec caching and same_names not True for zip files
+        if 'cache' in self.urlpath and 'zip' in self.urlpath:
+            same_names = False  # default
+            # find different same_names setting
+            for c in ['blockcache', 'filecache', 'simplecache']:
+                if c in self.storage_options:
+                    if 'same_names' in self.storage_options[c]:
+                        same_names = self.storage_options[c]['same_names']
+            if not same_names:
+                warnings.warn(
+                    'Need same_names = True for local caching of `zip` files.'
+                )
 
         super().__init__(metadata=metadata)
 
     def _open_dataset(self):
         """
         Open dataset using geopandas and use pattern fields to set new columns.
         """
+        if 'cache' in self.urlpath:  # new caching
+            url = fsspec.open_local(self.urlpath, **self.storage_options)
+            if not os.path.exists(url):
+                # explicitly loading the file if not cached
+                url = fsspec.open(self.urlpath, **self.storage_options)
+        elif self.cache:  # old caching
+            self.cache[0].load(self.urlpath)
+            if isinstance(self.cache[0], str):
+                url = self.cache[0]
+            else:
+                url = self.urlpath
+        else:  # no caching
+            url = self.urlpath
+
+        # opening locally cached files, geopandas expects local zip paths with `zip://`
+        # see https://geopandas.org/io.html
+        try:
+            if requests.head(url).status_code == 200:
+                is_remote = True
+            else:
+                is_remote = False
+        except (
+            requests.exceptions.MissingSchema,
+            requests.exceptions.InvalidSchema,
+            requests.exceptions.ConnectionError,
+        ):
+            is_remote = False
+        if not is_remote:
+            # extract ending, expect url
+            if '.' in self.urlpath.split('/')[-1]:
+                ending = self.urlpath.split('.')[-1]
+            else:
+                ending = None
+            if ending == 'zip':
+                url = f'{ending}://{url}'
```

Comment:
This feels brittle to me. Why not trust the catalog author to enter the correct URL (with fiona virtual filesystems if necessary) in the first place? Does this work with compound URLs?

Reply:
I tried to implement this with compound URLs, but didnt succeed. see https://gist.github.com/aaronspring/3a865b991f40fa7b3afef102df0717f8